### DISCUSSION TITLE
`into_make_service` implemented for `Router<(), B>`

### DISCUSSION THREAD

**Discussion by MidasLamb** [tokio-rs/axum#1602] [Q&A] [answered]
I've got a function that makes a router with a state, so it returned a `Router<AppState>`, however, it's not possible to call `.into_make_service()` and the result of the function.

I changed the return-type to `Router<()>` and now it works. However I don't understand how that's possible that it now silently turns a `Router<AppState>` into a `Router<()>`? Because I still call `.with_state(AppState)` which should convert it to a `Router<AppState>`.

It seems that `.into_make_service()` should be defined irrespective of the type of the State, rather than only being available for `Router<()>`

**Comment by davidpdrsn** [comment, marked as answer]
That is explained in the docs for [`Router::with_state`](https://docs.rs/axum/latest/axum/routing/struct.Router.html#method.with_state) specifically the "What `S` in `Router<S>` means" section

**Reply by MidasLamb** [reply]
Ah ok I missed that, thanks for the quick reply!
